src/darwin/Framework/CHIP/MTRDevice.h

```diff
@@ -201,6 +235,25 @@ typedef NS_ENUM(NSUInteger, MTRDeviceState) {
                                       completion:(MTRDeviceOpenCommissioningWindowHandler)completion
     MTR_AVAILABLE(ios(17.0), macos(14.0), watchos(10.0), tvos(17.0));
 
+/**
+ * Download log of the desired type from the device.
+ *
+ * Note: The consumer of this API should move the file that the NSURL in the MTRDiagnosticLogResult points to
+ * or open it for reading before the completion handler returns. Otherwise, the file will be deleted, and the data will be lost.
+ *
+ * @param type       The type of log being requested. This should correspond to a value in the enum MTRDiagnosticLogType.
+ * @param timeout    The timeout for getting the log. If the timeout expires, completion will be called with whatever
+ *                   has been retrieved by that point (which might be none or a partial log).
+ *                   If the timeout is set to 0, the request will not expire and completion will not be called until
+ *                   the log is fully retrieved or an error occurs.
+ * @param queue      The queue on which completion will be called.
+ * @param completion The completion that will be called to pass in the URL for the requested log.
+ */
+- (void)downloadLogOfType:(MTRDiagnosticLogType)type
+                  timeout:(NSTimeInterval)timeout
+                    queue:(dispatch_queue_t)queue
+               completion:(void (^)(MTRDiagnosticLogResult * _Nullable logResult, NSError * _Nullable error))completion MTR_PROVISIONALLY_AVAILABLE;
```

Comment:
Since we've gone back to just downloading one log, should the completion just get the relevant `NSURL` directly?

Reply:
Mostly depends on how we want to handle timeout, right?